shell/platform/android/test/io/flutter/embedding/android/FlutterViewTest.java

```diff
@@ -267,6 +270,77 @@ public void itSendsDarkPlatformBrightnessToFlutter() {
     assertEquals(SettingsChannel.PlatformBrightness.dark, reportedBrightness.get());
   }
 
+  @Test
```

Comment:
I only added tests for the `attachToFlutterEngine` callpath but can't find a good way to test when the setting was changed while the app is running. Any suggestions?

Reply:
Are we able to mock systemSettingsObserver?  Probably more trouble than it's worth.

Reply:
There's a significant delay between the setting value change the and the notification fire, I think that's one thing that made the testing this hard.